dev_guide/cicd.adoc

```diff
@@ -0,0 +1,254 @@
+
+For your source code (in the classic sense), the OpenShift `*Build*`, `*BuildRequest*`, and `*BuildConfig*` object are of course the starting point.  They expose the most typically used
+features of the `*SCM*` (most notably Git) to access the appropriate version of your source code (commit IDs, branches, context directories, etc.) and build the source code.
+
+But source code management (`*SCM*`) tooling is also leveraged througout the industry for managing many artifacts beyond what classically is defined as source code.
+OpenShift also promotes such a strategy.
+
+And where a `*SCM*` is not appropriate, most notably Images, we have the OpenShift Internal Image Registry and `*ImageStreams*`. 
+
+[[api-objects]]
+=== API Objects
+
+As is highlighted throughout OpenShift documentation, every API Object can be expressed via either JSON or YAML.  That JSON or YAML can be stored in and retrieved from a
+`*SCM*` system, with the its versioning related capabilities, including the creation of branches, assignment of and query on various labels or tags associated to versions.
+
+Also, the API Ojects are designed such that there are portions of the object which are intended to *specify* the Object's intent, and other
+portions which reflect the *status* or current state of the Object.  Inputs and outputs in other words.  The input portions, when expressed in JSON/YAML, in particular are items
+that naturally fit as artifacts stored in a `*SCM*`.
+
+[NOTE]
+Remember, the input or specification portions of the API Objects can be totally static, or dynamic in the sense that variable substitution via template processing is possible on actual instantiation.
+
+More on all this later when we dive into the more precise examples.
+
+[[s2i]]
+=== Image construction
+
+The next re-introduction of previously discussed items in OpenShift documentation: the Dockerfiles or S2I scripts and config files used to compose images are "source" in every sense, and typically
+managed via souce code management tooling in `*SCM*` repositories as well.  Tag, label, and manage versions of those artifacts.  Check out varying versions of those artifacts during the build jobs
+of your `*CI/CD*` tooling if it makes sense to vary the image composition in that way at different points of your `*CI/CD*` pipeline (though admittedly, such manipulations are most likely to occur in
+the earlier, development type stages). 
+
+[[image-streams]]
+=== Images themselves and ImageStreams
+
+Again, as noted earlier,  Images are now artifacts of your application.  And in some cases, an Image (especially one produced by an OpenShift `*Build*`) might encapsulate the entirety of your application.
+The new application binary if you will.
+
+To date, Images are not managed in a `*SCM*`itself (just like binaries were not).  However, just as installable artifacts and repositories (i.e. RPMs, RPM repo's) arose with similar semantics to `*SCMs*`,
```

Comment:
s/managed in a SCM itself/managed in an SCM system/

suggest: just as the need to manage binary build artifacts in repositories has gained importance (e.g. maven), so to has the need to manage and curate container images.